### PR TITLE
Adding pr comment for asset size report

### PR DESCRIPTION
**Pull request by patrickhousley**
### Overview

Updating CI to generate an asset size comparison between a PR and the current dev build from main. This will post a comment to the PR with the size comparison table containing the minified loader assets.

### Related Issue(s)

https://issues.newrelic.com/browse/NEWRELIC-6740

### Testing

Let CI workflow run to test out the comment generation.

### CODE REVIEW

File: tools/scripts/diff-sizes.mjs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    { name: 'nr-loader-full', matcher: /nr-loader-full(?:-\d*?)?\.min\.js/ },
+    { name: 'nr-loader-rum', matcher: /nr-loader-rum(?:-\d*?)?\.min\.js/ }
+  ],
+  polyfills: [
+    { name: 'nr-loader-spa-polyfills', matcher: /nr-loader-spa-polyfills(?:-\d*?)?\.min\.js/ },
+    { name: 'nr-loader-full-polyfills', matcher: /nr-loader-full-polyfills(?:-\d*?)?\.min\.js/ },
+    { name: 'nr-loader-rum-polyfills', matcher: /nr-loader-rum-polyfills(?:-\d*?)?\.min\.js/ }
+  ],
+  worker: [
+    { name: 'nr-loader-worker', matcher: /nr-loader-worker(?:-\d*?)?\.min\.js/ }
```
Comment (cwli24) [inline tools/scripts/diff-sizes.mjs:29]: Could you append `.min` to these names for clarity?

Comment (patrickhousley) [reply tools/scripts/diff-sizes.mjs]: done

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,6 +7,36 @@ on:
...
+        run: npm ci
+      - name: Running Prod Build
+        run: npm run cdn:build:dev
+      - name: Generating Asset Size Report
+        run: node ./tools/scripts/diff-sizes.mjs -o build
+      - name: Creating PR Comment
+        run: |
+          node ./tools/scripts/comment-pr.mjs \
+            --pull-request=${{ github.event.pull_request.number }} \
+            --token=${{ secrets.GITHUB_TOKEN }} \
```
Comment (metal-messiah) [inline .github/workflows/ci.yml:36]: I can never quite remember the safety issues around these CI tasks, so I have to ask to make sure ;P, 

Is there any danger to using/exposing this in a `pull_request` type action?

Comment (patrickhousley) [reply .github/workflows/ci.yml:36]: If the CI workflow runs on a PR from a fork, the secrets will be pulled from that fork and not our repo. Our secrets cannot be exposed via a fork unless we use `pull_request_target`

**Review by metal-messiah — APPROVED** [review]

File: tools/scripts/diff-sizes.mjs [hunk 3]
```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (bjfield) [inline tools/scripts/diff-sizes.mjs:6]:
```suggestion
/*
 * Copyright 2020 New Relic Corporation. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
 
/**
 * @file Compares local build output files with those of a specified release version on CDN and writes JSON and
 * markdown build size reports to a specified directory.
 */
```

File: tools/scripts/comment-pr.mjs [hunk 4]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (bjfield) [inline tools/scripts/comment-pr.mjs:7]:
```suggestion

/**
 * @file Adds the provided input text as a comment to a specified Github pull request, or if an optional tag string is
 * provided, updates an existing comment containing that tag.
 */

```

**Review by bjfield — APPROVED** [review]
I've run some scripts and looked at the output. A nice addition! I've suggested some optional code comments, which you can commit if you think they are helpful.

**Pull request merged**